cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -37,20 +38,17 @@ type SourcegraphListResult struct {
 	// Note: this has a side-effect of setting a the "config fingerprint". The
 	// config fingerprint means we only calculate index options for repositories
 	// that have changed since the last call to IterateIndexOptions. If you want
-	// to force calculation of index options use Sourcegraph.GetIndexOptions.
+	// to force calculation of index options use
+	// Sourcegraph.ForceIterateIndexOptions.
 	IterateIndexOptions func(func(IndexOptions))
 }
 
 type Sourcegraph interface {
 	List(ctx context.Context, indexed []uint32) (*SourcegraphListResult, error)
 
 	// ForceIterateIndexOptions will best-effort calculate the index options for
-	// all of ids. If any repository fails it internally logs.
-	ForceIterateIndexOptions(func(IndexOptions), ...uint32)
-
-	// GetIndexOptions is deprecated but kept around until we improve our
-	// forceIndex code.
-	GetIndexOptions(repos ...uint32) ([]indexOptionsItem, error)
+	// all repos. For each repo it will call either onSuccess or onError.
+	ForceIterateIndexOptions(onSuccess func(IndexOptions), onError func(uint32, error), repos ...uint32)
```

Comment:
I like this interface! But why the prefix "Force"?

Reply:
The other method may not actually check for every repository you pass in. The Force method will always make a call for each repo passed in against gitserver/etc. I'll make the docstring clearer.